### PR TITLE
Error handling for Metadata API

### LINKED ISSUE #7: Account for metadata API not always being available
Either a timeout issue or a timing issue (API not started yet).

### PR DESCRIPTION
**Pull request by dylanfareed**
This PR:

* lightens the dependency load here: uses httpc rather than hackney, thinly wraps xmerl to parse XML responses from AWS describe instance requests, doesn't us ex_aws at all (we opt instead to sign urls to fetch data from AWS about other EC2 instances tagged as being in same cluster).
* provides error handling such that if any of the requests fail `Peerage.Via.Ec2.poll/0` will return an empty list rather than crashing.

Closes #7

### CODE REVIEW

File: lib/peerage/via/ec2.ex [hunk 1]
```diff
@@ -0,0 +1,106 @@
+defmodule Peerage.Via.Ec2 do
+  @moduledoc """
+  A Peerage provider for easy clustering on AWS EC2 and Elastic Beanstalk.
+  """
```
Comment (ksheurs) [inline lib/peerage/via/ec2.ex]: Being we're requiring the `peerage` dependency now, we could include https://github.com/mrluc/peerage/blob/master/lib/peerage/dns.ex#L2.

File: test/support/fixtures/fetch_running_services.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+                        <item>
+                            <key>service</key>
+                            <value>falkner</value>
+                        </item>
+                    </tagSet>
+                </item>
+            </instancesSet>
+        </item>
+    </reservationSet>
+</DescribeInstancesResponse>
```
Comment (ksheurs) [inline test/support/fixtures/fetch_running_services.xml]: Curious did you perform a request and copy/paste this or is this pulled from AWS docs? or something else?

Comment (dylanfareed) [reply test/support/fixtures/fetch_running_services.xml]: I'm using these fixtures in tests (forthcoming). 

Yep. Performed the requests and wrote the output to test/support/fixtures. I cleaned out the data irrelevant to this code. So this is greatly simplified for the tests.

File: lib/peerage/via/ec2.ex (hunk 1 above)
Comment (dylanfareed) [reply lib/peerage/via/ec2.ex]: Will update

Comment (dylanfareed) [reply lib/peerage/via/ec2.ex]: updated

File: test/peerage/via/ec2_test.exs [hunk 3]
```diff
@@ -0,0 +1,81 @@
+defmodule Peerage.Via.Ec2Test do
+  use ExUnit.Case, async: false
+
+  import Mock
+  alias Peerage.Via.Ec2
+  alias Peerage.Via.Ec2.SignedUrl.RequestTime
+
+  @now ~N[1980-10-17 01:23:45]
```
Comment (dylanfareed) [inline test/peerage/via/ec2_test.exs]: 🎂

**Pull request merged**